### PR TITLE
Add missing translations for dossier export.

### PR DESCRIPTION
**Pull request by 2e12**
A French translation isn't available when exporting the dossier cover, therefore it is added.

Example files:
[dossiercover.pdf](https://github.com/4teamwork/opengever.core/files/5274165/dossiercover.pdf)
[dossierdetails.pdf](https://github.com/4teamwork/opengever.core/files/5274166/dossierdetails.pdf)

Jira: https://4teamwork.atlassian.net/browse/GEVER-832

Definition of Done: https://4teamwork.atlassian.net/wiki/spaces/CHX4TW/pages/917562/

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- API change:
  - [ ] Documentation is updated
  - If breaking:
    - [ ] api-change label added
    - [ ] Scrum master is informed
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- DB-Schema migration
  - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
  - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

### CODE REVIEW

File: opengever/latex/dossiercover.py [hunk 1]
```diff
@@ -35,21 +37,55 @@ class DossierCoverLaTeXView(MakoLaTeXView):
...
-            'referencenr': self.convert_plain(self.get_referencenumber()),
-            'title': self.convert_plain(self.context.Title()),
-            'description': self.convert(self.get_description()),
-            'responsible': self.convert_plain(self.get_responsible()),
-
-            'start': self.convert_plain(self.context.toLocalizedTime(
-                    str(IDossier(self.context).start)) or '-'),
-            'end': self.convert_plain(self.context.toLocalizedTime(
-                    str(IDossier(self.context).end)) or '-'),
+        return self.get_dossier_metadata()
```
Comment (njohner) [inline opengever/latex/dossiercover.py:40]: I'm not a fan of this method that just proxies to another method. What's the point?

File: opengever/latex/dossiercover.py [hunk 2]
```diff
@@ -35,21 +37,55 @@ class DossierCoverLaTeXView(MakoLaTeXView):
...
+
+    def get_metadata_config(self):
```
Comment (njohner) [inline opengever/latex/dossiercover.py:42]: The naming is a bit weird to me. How is that a config?

File: opengever/latex/dossiercover.py [hunk 3]
```diff
@@ -35,21 +37,55 @@ class DossierCoverLaTeXView(MakoLaTeXView):
...
+
+    def get_dossier_metadata(self):
+        args = {}
+        metadata = self.get_metadata_config()
+        for key, item in metadata.items():
+            item['label'] = translate(item['label'], context=self.request)
+            if item['label'] == '':
+                args[key] = item['value']
+            else:
+                args[key] = u'{}: {}'.format(item['label'], item['value'])
```
Comment (njohner) [inline opengever/latex/dossiercover.py:88]: I find that whole thing pretty convoluted. First you set labels for all the metadata, then when it's an empty string you get rid of it.
Moreover I feel like you are doing the template's job here. Why not simply add the few missing labels to the original missing `args` dictionary you had? You did not make the code less complex and you did not make the code shorter, so what did you gain?
I'd either simply add 'responsible_label', `start_label` and so on to `args` or I'd go maybe for something like:
```
    translatable_labels = {
        'responsible_label': _('label_responsible', default='Responsible')
    }
    def get_render_arguments(self):
        args = self.get_dossier_metadata()
        args.update(self.get_metadata_labels())
  
    def get_translated_labels(self):
        args = {}
        for key, label in self.translatable_labels.items():
              args[key] = translate(label, context=self.request)
        return args
```

But again the simplest solution of adding the needed labels in `args` might even be easier to read

File: opengever/latex/templates/dossiercover.tex [hunk 4]
```diff
@@ -15,16 +15,16 @@
 %% \end{tabularx}
 \vspace{0.5cm}
 \rule{\linewidth}{.11em}
-Version Ordnungssystem: ${repositoryversion}\newline
-\textbf{Aktenzeichen: ${referencenr}}\newline
+${repositoryversion}\newline
+\textbf{${referencenr}}\newline
```
Comment (njohner) [inline opengever/latex/templates/dossiercover.tex:19]: and here you would have:
`\textbf{${referencenr_label}: ${referencenr}}\newline`

File: opengever/latex/tests/test_default_layout.py [hunk 5]
```diff
@@ -83,94 +43,69 @@ def test_before_render_hook(self):
...
+            'page_label': 'Page',
+            'client_title': 'Hauptmandant',
+            'location': 'Bern',
             'show_organisation': False,
-            'location': FAKE_LOCATION,
+            'member_phone': '012 345 6789',
+            'show_logo': False
             }
 
         self.assertEqual(layout.get_render_arguments(), args)
```
Comment (njohner) [inline opengever/latex/tests/test_default_layout.py:67]: `assertDictEqual` would be better

**Review by njohner — CHANGES_REQUESTED** [review]
Over LGTM, but I think you took a bit of a convoluted approach, see my comment. Also, try to make your commit message titles a bit more explicit. You also have some typos in the commit messages, notably `is'nt` should be `isn't`

**Comment by 2e12** [comment]
@njohner Thank you for your feedback.I tried to make it as similar as possible to [dossierdetails.py,](https://github.com/4teamwork/opengever.core/blob/master/opengever/latex/dossierdetails.py) which is why the code is the way it is. But I can understand your feedback and I will adapt my changes accordingly.

**Review by njohner — APPROVED** [review]
LGTM

**Pull request merged**